packages/vue-urql/src/utils.ts

```diff
@@ -1,45 +1,150 @@
-import type { GraphQLRequest, AnyVariables } from '@urql/core';
-import type { Ref, ShallowRef } from 'vue';
-import { isRef } from 'vue';
+import type {
+  AnyVariables,
+  Client,
+  CombinedError,
+  DocumentInput,
+  Operation,
+  OperationContext,
+  OperationResult,
+  OperationResultSource,
+} from '@urql/core';
+import { createRequest } from '@urql/core';
+import type { Ref } from 'vue';
+import { watchEffect, isReadonly, computed, ref, shallowRef, isRef } from 'vue';
+import type { UseSubscriptionArgs } from './useSubscription';
+import type { UseQueryArgs } from './useQuery';
 
 export type MaybeRef<T> = T | (() => T) | Ref<T>;
 export type MaybeRefObj<T> = T extends {}
   ? { [K in keyof T]: MaybeRef<T[K]> }
   : T;
 
-export const unref = <T>(maybeRef: MaybeRef<T>): T =>
+const unwrap = <T>(maybeRef: MaybeRef<T>): T =>
   typeof maybeRef === 'function'
     ? (maybeRef as () => T)()
     : maybeRef != null && isRef(maybeRef)
     ? maybeRef.value
     : maybeRef;
 
-export interface RequestState<
-  Data = any,
-  Variables extends AnyVariables = AnyVariables,
-> {
-  request: GraphQLRequest<Data, Variables>;
-  isPaused: boolean;
-}
+const unwrapDeeply = <T>(input: T): T => {
+  input = isRef(input) ? (input.value as T) : input;
 
-export function createRequestState<
-  Data = any,
-  Variables extends AnyVariables = AnyVariables,
->(
-  request: GraphQLRequest<Data, Variables>,
-  isPaused: boolean
-): RequestState<Data, Variables> {
-  return { request, isPaused };
-}
+  if (typeof input === 'function') {
+    return unwrapDeeply(input()) as T;
+  }
 
-export const updateShallowRef = <T extends Record<string, any>>(
-  ref: ShallowRef<T>,
-  next: T
-) => {
-  for (const key in next) {
-    if (ref.value[key] !== next[key]) {
-      ref.value = next;
-      return;
+  if (input && typeof input === 'object') {
+    const isArray = Array.isArray(input);
+    const out = (isArray ? [] : {}) as T;
+    for (const prop in input) {
+      if (
+        isArray ||
+        (Object.hasOwn || Object.prototype.hasOwnProperty.call)(input, prop)
```

Comment:
This polyfill will fail because `.call()` is a virtual method.
```
> (false || Object.prototype.hasOwnProperty.call)({}, 'test')

TypeError: (false || Object.prototype.hasOwnProperty.call) is not a function
    at <anonymous>:1:48
```

Reply:
I think a good option would be to basically remove `Object.hasOwn` for now. Plus it is slower according to this benchmark https://jsperf.app/labega/2.
![image](https://github.com/user-attachments/assets/37654b2b-07f1-47e4-9ba4-746216a084df)

Reply:
thanks, removed